filter/challenges-08.test.js

```diff
@@ -0,0 +1,287 @@
+'use strict';
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 1
+//
+// Write a function named oddValues that, given an array of integers as input,
+// uses filter to return an array containing only the odd integers.
+//
+// For example, oddValues([1,2,3]) returns [1,3].
+// ------------------------------------------------------------------------------------------------
+
+const oddValues = (input) => {
+  return input.filter(values => values % 2 !== 0)
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 2
+//
+// Write a function named filterStringsWithVowels that, given an array of strings as input,
+// uses filter to return an array with only words that contain vowels.
+//
+// The callback function to filter should include a regular expression pattern.
+//
+// For example, filterStringsWithVowels(['gregor','hound','xyz']) returns ['gregor', 'hound'].
+// ------------------------------------------------------------------------------------------------
+
+
+const filterStringsWithVowels = (input) => {
+  let regex = (/[aeiou]/i);
+  return input.filter(value => regex.test(value));
+};
+
+
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 3
+//
+// Write a function named notInFirstArray that, given two arrays as input, uses filter to return
+// an array of all the elements in the second array that are not included in the first array.
+//
+// For example, notInFirstArray([1,2,3], [1,2,3,4]) returns [4].
+// ------------------------------------------------------------------------------------------------
+
+const notInFirstArray = (forbiddenValues, input) => {
+  let allowedValues = input.filter(value => {
+    if (forbiddenValues.indexOf(value) < 0){
+      return true;
+    }

```

Comment:
I would include the `else {return false;}` lines as well, to make it clear what the behavior of this filter callback is.